code-experiments/build/java/ExampleExperiment.java

```diff
@@ -182,6 +189,8 @@ public static void myRandomSearch(Function f,
 		    /* Call the evaluate function to evaluate x on the current problem (this is where all the COCO logging
 		     * is performed) */
 			y = f.evaluate(x);
+			if (numberOfConstraints > 0)
+				z = f.evaluateConstraint(x);
```

Comment:
I don't think there should or need to be any prior expectation, because for many algorithms the order does not matter at all.